flink-yarn/src/test/java/org/apache/flink/yarn/YarnApplicationMasterRunnerTest.java

```diff
@@ -109,4 +120,37 @@ public Object answer(InvocationOnMock invocationOnMock) throws Throwable {
 			taskManagerConf, workingDirectory, taskManagerMainClass, LOG);
 		assertEquals("file", ctx.getLocalResources().get("flink.jar").getResource().getScheme());
 	}
+
+	@Test
+	public void testRunAndInitializeFileSystem() throws Exception {
+		// Mock necessary system variables
+		Map<String, String> map = new HashMap<String, String>(System.getenv());
+		map.put(YarnConfigKeys.ENV_HADOOP_USER_NAME, "foo");
+		// Create dynamic properties to be used in the Flink configuration
+		map.put(YarnConfigKeys.ENV_DYNAMIC_PROPERTIES, "myKey=myValue");
+		CommonTestUtils.setEnv(map);
+
+		// Create a temporary flink-conf.yaml and to be deleted on JVM exits
+		File currDir = new File(System.getenv().get(ApplicationConstants.Environment.PWD.key()));
+		String path = String.format("%s/%s.%s", currDir, "flink-conf", "yaml");
+		File f = new File(path);
+		f.createNewFile();
+		f.deleteOnExit();
+
+		// Mock FileSystem.initialize()
+		PowerMockito.mockStatic(FileSystem.class);
+		PowerMockito.doNothing().when(FileSystem.class);
+		FileSystem.initialize(any(Configuration.class));
+
+		String[] args = new String[5];
+		YarnApplicationMasterRunner yarnApplicationMasterRunner = new YarnApplicationMasterRunner();
+		yarnApplicationMasterRunner.run(args);
```

Comment:
I think we should refactor the code like YarnTaskManagerRunnerFactoryTest. Running the entire appMaster code in unittest is too much.